internal/dag/builder.go

```diff
@@ -630,6 +637,33 @@ func (b *Builder) computeRoutes(sw *ObjectStatusWriter, proxy *projcontour.HTTPP
 	return routes
 }
 
+func includeConditionsIdentical(includes []projcontour.Include) bool {
+	j := 0
+	prefixesMatch := false
+	headersMatch := false
+	for i := 1; i < len(includes); i++ {
+		// Now compare each include's set of conditions
+		for _, cA := range includes[i].Conditions {
+			for _, cB := range includes[j].Conditions {
```

Comment:
Nit: You could now simplify the inner check:
```Go
if (cA.Prefix == cB.Prefix) && cmp.Equal(cA.Header, cB.Header) {
    return true
}
```

Reply:
Nice! I tend to make things simpler which makes the code read on and on, this makes it more clear. =)